libguard/guard_interface.cpp

```diff
@@ -333,10 +372,12 @@ static void invalidateRecord(const guardRecordParam& value)
              (existGuard.targetId == entityPath)) &&
             (existGuard.recordId != GUARD_RESOLVED))
         {
+            toggleWriteFlag(file);
```

Comment:
this is not correct, these chain operatoins are dangerous you look for CustoFD class you need to follow that, constructor of CustomFd opens the file and destructor closes the fd. 

{
CustomFD fd;
.
.
}
when it reaches here fd will be closed